index.js

```diff
@@ -1,254 +1,263 @@
-          async.seq(sortComponents, removeUnstarted, stopComponents, function(cb) {
-              debug('System %s stopped', params.name)
-              running = false
-              cb()
-          })(function(err) {
-            if (err) return reject(err)
-            resolve();
-          })
+        const p = new Promise(function (resolve, reject) {
+            async.seq(sortComponents, removeUnstarted, stopComponents, function (cb) {
+                debug('System %s stopped', params.name)
+                running = false
+                cb()
+            })(function (err) {
+                if (err) return reject(err)
+                resolve();
+            })
         })
         return cb ? p.then(immediateCallback(cb)).catch(immediateError(cb)) : p
-      }
+    }
 
-    function stopComponents(components, cb) {
+    const stopComponents = (components, cb) => {
         async.eachSeries(components, stopComponent, cb)
     }
 
-    function stopComponent(name, cb) {
+    const stopComponent = (name, cb) => {
         debug('Stopping component %s', name)
-        var stop = definitions[name].component.stop || noop
-        var onStopped = function(err) {
+        const stop = definitions[name].component.stop || noop
+        const onStopped = (err) => {
             if (err) return cb(err)
             debug('Component %s stopped', name)
             setImmediate(cb)
         }
-        var p = stop(onStopped)
+        const p = stop(onStopped)
         if (p && p.then) {
-          p.then(immediateCallback(onStopped)).catch(immediateError(cb))
+            p.then(immediateCallback(onStopped)).catch(immediateError(cb))
         }
     }
 
-    function sortComponents(cb) {
-        var result = []
+    const sortComponents = (cb) => {
+        let result = []
         try {
-            var graph = new Toposort()
-            Object.keys(definitions).forEach(function(name) {
-                graph.add(name, map(definitions[name].dependencies, 'component'))
+            const graph = new Toposort()
+            Object.keys(definitions).forEach(function (name) {
+                graph.add(name, definitions[name].dependencies.map(dep => dep.component))
             })
-            result = intersection(graph.sort(), Object.keys(definitions))
+            result = arraysIntersection(graph.sort(), Object.keys(definitions))
         } catch (err) {
             return cb(err)
         }
         return cb(null, result)
     }
 
-    function removeUnstarted(components, cb) {
-        cb(null, intersection(components, started))
+    const removeUnstarted = (components, cb) => {
+        cb(null, arraysIntersection(components, started))
     }
 
-    function getDependencies(name, system, cb) {
-        async.reduce(definitions[name].dependencies, {}, function(accumulator, dependency, cb) {
+    const getDependencies = (name, system, cb) => {
+        async.reduce(definitions[name].dependencies, {}, function (accumulator, dependency, cb) {
```

Comment:
Prefer no space between `function` and `(params)`

Reply:
Prefer no `function` keyword xD, I missed that in the upgrade